library/src/androidTest/java/org/xmtp/android/library/DmTest.kt

```diff
@@ -400,4 +402,114 @@ class DmTest {
             assertEquals(dm.consentState(), ConsentState.ALLOWED)
         }
     }
+
+    @Test
+    fun testDmDisappearingMessages() = runBlocking {
+        val initialSettings = MessageDisappearingSettings(
+            1_000_000_000,
+            1_000_000_000 // 1s duration
+        )
+
+        // Create group with disappearing messages enabled
+        val boDm = boClient.conversations.findOrCreateDm(
+            alix.walletAddress,
+            messageDisappearingSettings = initialSettings
+        )
+        boDm.send("howdy")
+        alixClient.conversations.syncAllConversations()
+
+        val alixDm = alixClient.findDmByInboxId(boClient.inboxId)
+
+        // Validate messages exist and settings are applied
+        assertEquals(boDm.messages().size, 1) // howdy
+        assertEquals(alixDm?.messages()?.size, 1) // howdy
+        Assert.assertNotNull(boDm.messageDisappearingSettings)
+        assertEquals(boDm.messageDisappearingSettings!!.disappearDurationInNs, 1_000_000_000)
+        assertEquals(boDm.messageDisappearingSettings!!.disappearStartingAtNs, 1_000_000_000)
+        Thread.sleep(5000)
+        // Validate messages are deleted
+        assertEquals(boDm.messages().size, 0)
+        assertEquals(alixDm?.messages()?.size, 0)
+
+        // Set message disappearing settings to null
+        boDm.updateMessageDisappearingSettings(null)
+        boDm.sync()
+        alixDm!!.sync()
+
+        assertNull(boDm.messageDisappearingSettings)
+        assertNull(alixDm.messageDisappearingSettings)
+
+        // Send messages after disabling disappearing settings
+        boDm.send("message after disabling disappearing")
+        alixDm.send("another message after disabling")
+        boDm.sync()
+
+        Thread.sleep(1000)
+
+        // Ensure messages persist
+        assertEquals(
+            boDm.messages().size,
+            2
+        ) // disappearing settings 1, disappearing settings 2, boMessage, alixMessage
```

Comment:
All these counts will need to be updated once we start showing group update messages in the dms. @mchenani